osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs

```diff
@@ -463,6 +463,46 @@ public void TestHideSetSelectsCorrectBeatmap()
             AddAssert("Selected beatmap has not changed", () => songSelect.Carousel.SelectedBeatmap.ID == previousID);
         }
 
+        [Test]
+        public void TestDifficultyIconSelecting()
+        {
+            int? previousID = null;
+            addRulesetImportStep(0);
+            createSongSelect();
+
+            AddStep("Store current ID", () => previousID = songSelect.Carousel.SelectedBeatmap.ID);
+            AddStep("Click on a difficulty", () =>
+            {
+                InputManager.MoveMouseTo(songSelect.Carousel.ChildrenOfType<DrawableCarouselBeatmapSet.FilterableDifficultyIcon>()
+                                                   .First(icon => icon.Item.Beatmap != songSelect.Carousel.SelectedBeatmap));
+
+                InputManager.PressButton(MouseButton.Left);
+                InputManager.ReleaseButton(MouseButton.Left);
+            });
+            AddAssert("Selected beatmap changed", () => songSelect.Carousel.SelectedBeatmap.ID != previousID);
```

Comment:
This assertion is a bit strange; IMO it should be checking that the beatmap corresponding to the actual item clicked is selected, as that's the desired behaviour here really.